processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -685,6 +692,15 @@ public int size()
     return numEntries.get();
   }
 
+  public int estimatedOccupation()
+  {
+    int occupation = maxLengthForAggregators * getFacts().size();
```

Comment:
I assume this is meant to check the heap occupation. This is a great change, it's not awesomely accurate but it's probably better than guessing at maxRowsInMemory.

It'd be good to have a comment about the ways in which the estimate can be inaccurate:

- maxLengthForAggregators is based on max offheap size, which may not apply to onheap aggregations.
- dimension estimation is based on dictionary size only, not facts.
- likely to be an underestimate rather than an overestimate.

Also, the heap footprint is really different depending on which IncrementalIndex implementation is being used, so maybe it's best to have IncrementalIndex do `return 0` and move this impl to OnheapIncrementalIndex. I think the other incremental index impls would never have this method called (because Sink always uses OnheapIncrementalIndex).